src/utils/proxySet.ts

```diff
@@ -0,0 +1,96 @@
+import { proxy } from 'valtio'
+
+// properies that we don't want to expose to the end-user
+interface InternalProxySet<T> extends Set<T> {
+  data: T[]
+  toJSON: object
+  hasProxy(value: T): boolean
+}
+
+export function proxySet<T>(
```

Comment:
In this case, it's controversial, but currently we don't use function declaration in valtio (in jotai, we prefer function declaration, so we might change it eventually.)
```suggestion
export const proxySet = <T>(
```